src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CallVariantsFromAlignedContigsSpark.java

```diff
@@ -0,0 +1,382 @@
+            fullName = "minAlignLength", optional = true)
+    private Integer minAlignLength = CallVariantsFromAlignedContigsSpark.DEFAULT_MIN_ALIGNMENT_LENGTH;
+
+    // This class requires a reference parameter in 2bit format (to broadcast) and a reference in FASTA format
+    // (to get a good sequence dictionary).
+    // todo: document this better
+    @Argument(doc = "FASTA formatted reference", shortName = "fastaReference",
+            fullName = "fastaReference", optional = false)
+    private String fastaReference;
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+        Broadcast<ReferenceMultiSource> broadcastReference = ctx.broadcast(getReference());
+
+        final JavaRDD<AlignmentRegion> inputAlignedContigs = ctx.textFile(inputAlignments).map(AlignAssembledContigsSpark::parseAlignedAssembledContigLine);
+
+        final JavaPairRDD<Tuple2<String, String>, Iterable<AlignmentRegion>> alignmentRegionsKeyedByBreakpointAndContig = inputAlignedContigs.mapToPair(alignmentRegion -> new Tuple2<>(new Tuple2<>(alignmentRegion.assemblyId, alignmentRegion.contigId), alignmentRegion)).groupByKey();
+
+        final JavaPairRDD<String, ContigsCollection> assemblyIdsToContigCollections = loadContigsCollectionKeyedByAssemblyId(ctx, inputAssemblies).cache();
+
+        final JavaPairRDD<Tuple2<String, String>, byte[]> contigSequences = assemblyIdsToContigCollections.flatMapToPair(assemblyIdAndContigsCollection -> {
+            final String assemblyId = assemblyIdAndContigsCollection._1;
+            final ContigsCollection contigsCollection = assemblyIdAndContigsCollection._2;
+            final List<Tuple2<Tuple2<String, String>, byte[]>> contigSequencesKeyedByAssemblyIdAndContigId =
+                    contigsCollection.getContents().stream().map(pair -> new Tuple2<>(new Tuple2<>(assemblyId, pair._1.toString()), pair._2.toString().getBytes())).collect(Collectors.toList());
+            return contigSequencesKeyedByAssemblyIdAndContigId;
+        });
+
+        final JavaPairRDD<Tuple2<String, String>, Tuple2<Iterable<AlignmentRegion>, byte[]>> alignmentRegionsWithContigSequences = alignmentRegionsKeyedByBreakpointAndContig.join(contigSequences);
+
+        final Integer minAlignLengthFinal = this.minAlignLength;
+        final JavaRDD<VariantContext> variantContexts = callVariantsFromAlignmentRegions(broadcastReference, alignmentRegionsWithContigSequences, minAlignLengthFinal);
+
+        writeVariants(fastaReference, logger, variantContexts, getAuthenticatedGCSOptions(), outputPath);
+
+    }
+
+    protected static JavaRDD<VariantContext> callVariantsFromAlignmentRegions(final Broadcast<ReferenceMultiSource> broadcastReference, final JavaPairRDD<Tuple2<String, String>, Tuple2<Iterable<AlignmentRegion>, byte[]>> alignmentRegionsWithContigSequences, final Integer minAlignLengthFinal) {
```

Comment:
As this becomes the workflow manager of the actual variant calling, it's worth it to document the input arguments, especially the second one (it's long and the first Tuple2 is not obviously telling its purpose):
`final JavaPairRDD<Tuple2<String, String>, Tuple2<Iterable<AlignmentRegion>, byte[]>> alignmentRegionsWithContigSequences`